tests/components/UserActions/QueryList.spec.ts

```diff
@@ -110,6 +112,42 @@ describe('QueryList', () => {
         });
     });
 
+    it('should not show the same query twice', () => {
+        const TEST_DUPLICATE = [
+            new UserAction(UserActionType.Search, new Date(0), { query_expression: 'someQuery', origin_level_1: 'foo' }, null, 'someQuery'),
+            new UserAction(UserActionType.Search, new Date(1), { query_expression: 'someQuery2', origin_level_1: 'foo' }, null, 'someQuery2'),
+            new UserAction(UserActionType.Search, new Date(2), { query_expression: 'someQuery2', origin_level_1: 'foo' }, null, 'someQuery2'),
+            new UserAction(UserActionType.Search, new Date(3), { query_expression: 'someQuery2', origin_level_1: 'foo' }, null, 'someQuery2'),
+            new UserAction(UserActionType.Search, new Date(4), { query_expression: 'someQuery', origin_level_1: 'foo' }, null, 'someQuery')
+        ];
+
+        const mock = Mock.advancedComponentSetup<QueryList>(
+            QueryList,
+            new Mock.AdvancedComponentSetupOptions(null, { userId: 'testuserId' }, env => {
+                fakeUserProfileModel(env.root, sandbox).getActions.returns(Promise.resolve(TEST_DUPLICATE));
+                return env;
+            })
+        );
+
+        return delay(() => {
+            // Expand the whole list of query.
+            const button = mock.cmp.element.querySelector<HTMLElement>('.coveo-more-less');
+            if (button) {
+                button.click();
+            }
+
+            const list = mock.env.element.querySelector<HTMLOListElement>('.coveo-list');
+            expect(list.childElementCount).toBe(new Set(TEST_DUPLICATE.map(action => action.query)).size);
+
+            // Check that the order is respected.
+            TEST_DUPLICATE.reverse()
```

Comment:
No, you're reproducing the logic of the code, thus the tests does not really test per se.
Hardcode the input and output test values and compare the actual output to the test output.

Reply:
ditto

Reply:
ditto ditto? dittoto ? dittotto ?